modules/collections/client/controllers/collections.controllers.js

```diff
@@ -159,145 +174,210 @@ collectionModules.controller('CollectionEditCtrl',
         });
       };
 
+      // Publish a collection.
       $scope.publish = function() {
-        var publishedMainDocs = _.filter($scope.collection.mainDocuments, function (item) {
-          return item.document.isPublished === true;
-        });
-
-        if (publishedMainDocs.length < 1) {
-          AlertService.error("At least one 'PUBLISHED' main document is required to publish the collection.", 4000, true);
+        $scope.checkValidity(true);
+        if ($scope.validationFlags.detailsTabInvalid || $scope.validationFlags.documentsTabInvalid) {
           return;
         }
-
-        $scope.confirmPublishView(true).result.then(function() {
-          $scope.collection.isPublished;
-          return CollectionModel.publishCollection($scope.collection._id);
-        })
+        $scope.confirmPublishView(true).result
           .then(function() {
-            // published, show the message, and go to list...
+            return CollectionModel.save($scope.collection);
+          })
+          .then(function() {
+            return $scope.submitDocs(false);
+          })
+          .then(function() {
+            $scope.collection.isPublished;
```

Comment:
You may want to wait to set the "published" flag until the collection was actually published on the database (e.g.: when the promise returned by `CollectionModel.publishCollection($scope.collection._id)` is fulfilled).
Setting the state here could lead to weird situations IF the collection is not published for some reason.

Reply:
I'm actually going to remove this altogether, as I believe `CollectionModel.publishCollection($scope.collection._id);` sets it anyway